public/css/neznam-atproto-share-comments.css

```diff
@@ -0,0 +1,58 @@
+.wp-block-comments {

```

Comment:
It might be beneficial to copy layout of existing block comments. That way all themes that support block comments will have proper style applied.

Reply:
This one was a bit complicated. I had to make two versions, one that's for block themes and one for classic themes. I've tested it against WordPress' themes from twentytwelve through twentytwentyfive, and they all look very close to native comments. What do you think about this was of handling the DOM?

Reply:
This way of handling DOM is the problem. I might have formulated the problem the wrong way. The way you currently build the DOM is not customizable by user. Here we offer just one (or now two) possible layouts with some CSS that we either lift from existing themes or not. 

I am hoping that we might be able to use some default logic in WP to load the template for comments (might not be even possible), or at least offer a clean way for users to customize.

First one, not sure if there is a good standard way that themes adhere to and use that logic to create the comments with their styles.

Second, provide an option for theme builder to have a file in e.g. themes/your-theme/templates/bsky-comments.php that can be used to render the layout. If the file doesn't exist then use 'our' style. 

Both of these are not necessary to deploy this, but it might be something to consider because users will want to customize this, and I would hate to force users to user !important to override our styles.

Reply:
@mbanusic Before diving into how to fix this, I also assumed that modern WP themes had a consistent DOM for comments and all design was adjusted within the CSS. Unfortunately, that turned out to not be the case. Across all the year-named themes, as well as the most popular open-source themes, there's not a consistent template structure. Even the class naming isn't consistent, which surprised me more.

_(I even went down the rabbit hole of trying to get the PHP to render the markup of a default comment and interpret that into the JS, but that had more problems than benefits.)_

I like the idea of pulling in a theme partial, like Woo does, to make overriding our comment structure easy. I'll have to experiment a bit for securely handling this on the client-side without needing the overhead of something like Handlebars. I also like everything that lets me ditch the `style` attached to the elements, each one of those feels like I'm admitting defeat.

Does this look acceptable on the theme you've tested it against, or haven't you made it that far yet? (No rush.)

Reply:
I did check on same default theme as you. That is OK. Some kind of default style definitely needs to be provided, but an easy option to match you theme style is always welcome.

Most of my projects are using custom themes. So I would like an option to easily customize, even an option to disable loading plugin styles and just loading my CSS would be great (DOM can stay). No need to load plugin CSS if everything is in the theme CSS and also avoids the need to override styles with !important.